src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,214 @@
+
+delayed_requester = DelayedRequester(DELAY)
+image_store = image.ImageStore(provider=PROVIDER)
+
+DEFAULT_QUERY_PARAM = {
+    "has_images": 1,
+    "rights_type_permissive": 1,
+    "limit": LIMIT,
+    "offset": 0
+}
+
+
+def main():
+    logger.info('Begin: Brooklyn Museum API requests')
+    condition = True
+    offset = 0
+    image_count = 0
+    headers = {'api_key': API_KEY}
+
+    while condition:
+        query_param = _build_query_param(offset)
+        endpoint = '{0}object?has_images=1&rights_type_permissive=1&limit={1}&offset={2}'.format(ENDPOINT, LIMIT, offset)
+        response_json, total_images = _get_response(query_param, headers=headers, endpoint=endpoint)
+        if response_json is not None and total_images != 0:
+            objIDs = list(obj['id'] for obj in response_json['data'])
+            for obj in objIDs:
+                endpoint = '{0}object/{1}'.format(ENDPOINT, obj)
+                response = delayed_requester.get(endpoint, params=None, headers=headers)
+                result = _extract_response_json(response)
+                image_count = _handle_response(result, obj)
+            logger.info(f'Total images till now {image_count}')
+            offset += LIMIT
+        else:
+            logger.error('No more images to process')
+            logger.info('Exiting')
+            condition = False
+    image_count = image_store.commit()
+    logger.info(f'Total number of images received {image_count}')
+
+
+def _build_query_param(offset=0, default_query_param=DEFAULT_QUERY_PARAM):
+    query_param = default_query_param.copy()
+    query_param.update(
+        offset=offset
+    )
+    return query_param
+
+
+def _get_response(query_param, headers, endpoint=ENDPOINT, retries=RETRIES):
+    response_json, total_images = None, 0
+    for tries in range(retries):
+        response = delayed_requester.get(
+                    endpoint,
+                    params=None,
+                    headers=headers
+                    )
+        if response.status_code == 200 and response is not None:
+            try:
+                response_json = response.json()
+                total_images = len(response_json['data'])
+            except Exception as e:
+                logger.warning(f'response not captured due to {e}')
+                response_json = None
+            if response_json is not None and total_images is not None:
+                break
+
+        logger.info('Retrying \n'
+                    f'endpoint -- {endpoint} \t'
+                    f' with parameters -- {query_param} ')
+    if tries == retries-1 and ((response_json is None) or
+                               (total_images is None)):
+        logger.warning('No more tries remaining. Returning Nonetypes.')
+        return None, 0
+    else:
+        return response_json, total_images
+
+
+def _handle_response(object_data, _object_id):
```

Comment:
Please break this function into smaller pieces (e.g., refactor it into multiple smaller functions).  It'll make testing easier, and it makes the logic easier to follow and verify.